packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.ts

```diff
@@ -0,0 +1,190 @@
+  UnassignQualificationDto,
+  QualificationDto,
+} from './qualification.dto';
+import { QualificationEntity } from './qualification.entity';
+import { ErrorQualification } from '../../common/constants/errors';
+import { ControlledError } from '../../common/errors/controlled';
+import { QualificationRepository } from './qualification.repository';
+import { UserEntity } from '../user/user.entity';
+import { UserRepository } from '../user/user.repository';
+import { UserStatus, Role } from '../../common/enums/user';
+
+@Injectable()
+export class QualificationService {
+  private readonly logger = new Logger(QualificationService.name);
+
+  constructor(
+    private readonly qualificationRepository: QualificationRepository,
+    private readonly userRepository: UserRepository,
+  ) {}
+
+  public async createQualification(
+    createQualificationDto: CreateQualificationDto,
+  ): Promise<QualificationDto> {
+    const newQualification = new QualificationEntity();
+    newQualification.reference = createQualificationDto.reference;
+    newQualification.title = createQualificationDto.title;
+    newQualification.description = createQualificationDto.description;
+
+    if (createQualificationDto.expiresAt) {
+      const providedExpiresAt = new Date(createQualificationDto.expiresAt);
+
+      if (providedExpiresAt <= new Date()) {
+        this.logger.log(
+          ErrorQualification.InvalidExpiresAt,
+          QualificationService.name,
+        );
+        throw new ControlledError(
+          ErrorQualification.InvalidExpiresAt,
+          HttpStatus.BAD_REQUEST,
+        );
+      } else {
+        newQualification.expiresAt = providedExpiresAt;
+      }
+    }
+
+    try {
+      await this.qualificationRepository.save(newQualification);
+      return {
+        reference: newQualification.reference,
+        title: newQualification.title,
+        description: newQualification.description,
+        expiresAt: newQualification.expiresAt || null,
+      };
+    } catch (e) {
+      this.logger.log(ErrorQualification.NotCreated, QualificationService.name);
+      throw new ControlledError(
+        ErrorQualification.NotCreated,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+  }
+
+  public async getQualifications(): Promise<QualificationDto[]> {
+    try {
+      const qualificationEntities =
+        await this.qualificationRepository.getQualifications();
+
+      return qualificationEntities.map((qualificationEntity) => {
+        return {
+          reference: qualificationEntity.reference,
+          title: qualificationEntity.title,
+          description: qualificationEntity.description,
+          expiresAt: qualificationEntity.expiresAt || null,
+        };
+      });
+    } catch (error) {
+      this.logger.log(`Failed to fetch credentials: ${error.message}`);
+      return [];
+    }
+  }
+
+  async delete(reference: string): Promise<void> {
+    const result = await this.qualificationRepository.delete({ reference });
```

Comment:
I would suggest creating a new method to delete in BaseRepository and catch the QueryFailedError there in case of failure